COMPONENT-API.md

```diff
@@ -1,376 +1,864 @@
+
+## Management
+
+### CLI
+
+#### Development Mode
+
+The easiest way to develop and iteratively test is to use the `pd dev` command tol deploy a local file, attach to a component, and automatically update the component on each local save. To deploy a new component with `pd dev`, run:
+
+```bash
+pd dev <filename>
 ```
 
-You can reference [`this`](#referencing-this) within the `run` method. `this` refers to the component, and provides access to [props](#props), [methods](#methods), and Pipedream-provided functions like [`this.$emit`](#thisemit).
+To attach to an existing deployed component, run:
 
-You can view logs produced by the `run` method in the **LOGS** section of the Pipedream UI for the component, or using the `pd logs` CLI command:
+```bash
+pd dev [--dc <existing-deployed-component-id>] <file-or-name>
+```
+
+#### Deploy
+
+##### From Local Code
+
+To deploy a via CLI, use the `pd deploy` command. 
 
 ```bash
-pd logs <deployed-component-name>
+pd deploy <filename>
 ```
 
-If the `run` method emits events using `this.$emit`, you can access the events in the **EVENTS** section of the Pipedream UI for the component, or using the `pd events` CLI command:
+E.g.,
 
 ```bash
-pd events <deployed-component-name>
+pd deploy my-component.js
 ```
 
-## Referencing `this`
+##### From Pipedream Github Repo
 
-`this` refers to the component, and provides access to [props](#props), [methods](#methods), and Pipedream-provided functions like [`this.$emit`](#thisemit).
+You can explore the components available to deploy in [Pipedream's Github repo](https://github.com/PipedreamHQ/pipedream/tree/master/components).
 
-### Referencing props
+```bash
+pd deploy <github-url>
+```
 
-Props can be accessed using `this.<prop-name>`. For example, a `secret` prop:
+E.g.,
 
-```javascript
-props: {
-  secret: "string"
-},
+```bash
+pd deploy https://github.com/PipedreamHQ/pipedream/blob/master/components/http/http.js
 ```
 
-can be referenced using `this.secret`.
+##### From Any URL
 
-### Referencing methods
+```bash
+pd deploy <url-to-raw-code>
+```
 
-Methods can be accessed using `this.<method-name>`. For example, a `random` method:
+E.g.,
 
-```javascript
-methods: {
-  random() {
-    return Math.random()
-  },
-}
+```bash
+pd deploy https://raw.githubusercontent.com/PipedreamHQ/pipedream/master/components/http/http.js
 ```
 
-can be run like so:
+#### Update
 
-```javascript
-const randomNum = this.random();
+View the [CLI command reference](https://docs.pipedream.com/cli/reference/#command-reference).
+
+#### Delete
+
+View the [CLI command reference](https://docs.pipedream.com/cli/reference/#command-reference).
+
+### UI
+
+#### Deploy
+
+You can find any deploy curated components at https://pipedream.com/sources/new, or you can deploy code via the UI using following URL patterns.
```

Comment:
"You can find any deploy curated components" -> did you mean "find and deploy"?

Reply:
yes - thx!